assembly/nfa/matcher.ts

```diff
@@ -123,10 +123,27 @@ export class CharacterSetMatcher extends Matcher {
 
   matches(code: u32): bool {
     _code = code;
-    if (!this.negated) {
-      return this.matchers.some((m) => m.matches(_code));
-    } else {
-      return !this.matchers.some((m) => m.matches(_code));
+    let match = false;
+    for (let i = 0; i < this.matchers.length; i++) {
+      if (matches(this.matchers[i], code)) {
+        match = true;
+        break;
+      }
     }
+    // @ts-ignore
+    return <bool>(this.negated ^ match);
+  }
+}
+
+function matches(matcher: Matcher, code: u32): bool {
```

Comment:
Aha!! does this solve the issue highlighted here?

https://github.com/ColinEberhardt/assemblyscript-regex/blob/main/assembly/regexp.ts#L155-L179

I do hope so!

Reply:
Yeah I didn't dig into it more, but found that a switch over the instance was a sure thing.

Reply:
@MaxGraey might have got there first! #21